json-schema-generator/src/main/java/io/micronaut/jsonschema/generator/aggregator/DefinitionsAggregator.java

```diff
@@ -0,0 +1,128 @@
+/*
+ * Copyright 2017-2024 original authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.micronaut.jsonschema.generator.aggregator;
+
+import io.micronaut.core.annotation.Internal;
+import io.micronaut.jsonschema.generator.CodeGenerator;
+import io.micronaut.jsonschema.model.Schema;
+import io.micronaut.sourcegen.model.ClassTypeDef;
+import io.micronaut.sourcegen.model.TypeDef;
+import jakarta.inject.Singleton;
+
+import java.util.AbstractMap;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import static io.micronaut.core.util.StringUtils.capitalize;
+import static io.micronaut.jsonschema.generator.aggregator.TypeAggregator.getCamelCaseName;
+import static io.micronaut.jsonschema.generator.aggregator.TypeAggregator.getTypeDefFromJson;
+
+/**
+ * An aggregator for storing and accessing definitions and oneOf relations from json schema.
+ * Saves a map of definition reference to TypeDef, Boolean (boolean value is true if the type is a class/interface object.
+ * Saves a map of oneOf objects to keep in track on inheriting objects.
+ *
+ * @author Elif Kurtay
+ * @since 1.2
+ */
+@Internal
+@Singleton
+public class DefinitionsAggregator {
```

Comment:
Might be better to call this `GeneratorContext`, as it is different from what AnnotationsAggregator seems to do